### ISSUE TITLE
Design and implement a landing page for people coming in via the Firefox Developer Tools

### ISSUE THREAD

**Issue by denschub** [webcompat/webcompat.com#1534]
Soon™, we will have an option inside our GoFaster addon to add a link if we changed a site. This notification targets both the site developers and potential users, to make them aware that we changed something, and, even better, to make clear what we've changed and how they can disable our modifications.

A message in the DevTools console might look like:

<img width="763" alt="screen shot 2017-04-26 at 11 19 42" src="https://cloud.githubusercontent.com/assets/344777/25427830/0b3c19e0-2a74-11e7-8308-223c4c14a9d8.png">

As mentioned before, we need to be able to target two different groups of people:

* The users, who might discover the message and are interested in the cause and history of the modification we did.
* The sites developers who want to know the solution of given issue and a way to test the site without having our modifications in place.

While in theory we could have more than one site linked in the devtools (for example a wiki page and the actual webcompat issue), I do not think we should - it already looks cramped with one link in place and adding more text will not improve the readability of our console message. So I'd like us to have some kind of "landing page" for webcompat issues to link there.

Contents I'd like to see on the page:

* A very short introduction of who we are and what/why we do so people are not confused that we mess with their webpages.
* A link to the webcompat issue to explain the cause of the issue, what we are exactly modifying (user agents, CSS, JS, ...) and what we tried before putting overrides in place. (Note that this could be both a webcompat.com issue and a Bugzilla issue, so we need some kind of flexibility here)
* Information on how to disable the GoFaster overrides for testing. At the moment, this would only be toggling `extensions.webcompat.perform_ua_overrides` in `about:config` to false, but we should provide some step-by-step instructions.

Ideally, we have a somewhat short url, with the issue ID in it, like `http://webcompat.com/devtools-landing/1234` or `http://webcompat.com/devtools-landing/bz1234567` for Bugzilla issues.

Maybe something else. These are just ideas for further discussion.

**Comment by magsout** [comment]
@denschub it's a very good idea. I like it. 👍

**Comment by zoepage** [comment]
I like the idea a lot!

> The users, who might discover the message and are interested in the cause and history of the modification we did.
    The sites developers who want to know the solution of given issue and a way to test the site without having our modifications in place.

@denschub Those two points are related towards content, so we would either need to provide content for this or links to e.g. release notes, correct? 

> So I'd like us to have some kind of "landing page" for webcompat issues to link there.

So, we'd have one entry point (= one URI) and go from there?


> **Contents I'd like to see on the page:**
    A very short introduction of who we are and what/why we do so people are not confused that we mess with their webpages.

+1
    
> A link to the webcompat issue to explain the cause of the issue, what we are exactly modifying (user agents, CSS, JS, ...) and what we tried before putting overrides in place. (Note that this could be both a webcompat.com issue and a Bugzilla issue, so we need some kind of flexibility here)

How would you like to generate that information? A link passed in the console with a param passed through? Or anything else in mind? :)

> Information on how to disable the GoFaster overrides for testing. At the moment, this would only be toggling extensions.webcompat.perform_ua_overrides in about:config to false, but we should provide some step-by-step instructions.

I like the option for disabling GoFaster for testing. Would it be set to true when you restart the browser? Can imagine, people forget about enabling it again?


I really like the proposal and would be happy to help. 👍

**Comment by denschub** [comment]
> so we would either need to provide content for this or links to e.g. release notes, correct?

A link to the web-bug probably would be enough. 

> So, we'd have one entry point (= one URI) and go from there?

Ack.

> How would you like to generate that information?

As long as the link is not too long, sure. Something like `https://webcompat.com/issueinfo/1234/?type=css` or whatever would work. We only have a limited subset of possible categories and providing a general text ("We've adjusted some CSS values to make the site look right on your browser.") is fine. 

> Would it be set to true when you restart the browser?

Not the right issue to have this discussion, but yeah, we reset the setting on every browser reboot!

**Comment by zoepage** [comment]
> As long as the link is not too long, sure. Something like https://webcompat.com/issueinfo/1234/?type=css or whatever would work. We only have a limited subset of possible categories and providing a general text ("We've adjusted some CSS values to make the site look right on your browser.") is fine.

How do you want to generate them and where should they point to? Like, could you walk me through, so I get a better understanding of what you'd like to add and how to help building it? :)

**Comment by denschub** [comment]
> How do you want to generate them

What do you mean? The links? We'd generate them by hand and add them per individual override.

**Comment by denschub** [comment]
Just had a call with @zoepage to talk a bit about details and it's probably the best to simply integrate that into the existing webcompat.com page, including its design. However, thit would mean that the Firefox-only landing page would be designed using the webcompat.com colors and schemes and not using the Firefox branding. Given the recent discussions, we'd like to make sure that everyone is fine with it.

So... @adamopenweb @karlcow @magsout @miketaylr, any objections in using the webcompat.com theme for the planned Firefox-only landing page? :)

**Comment by karlcow** [comment]
Still objecting for the same reasons we mentioned earlier. 
For me it's not a question of layout/branding but a question of scope. This should live on MDN.

Opera
See http://www.opera.com/docs/browserjs/
and https://github.com/operasoftware/browserjs/

Microsoft
See https://msdn.microsoft.com/en-us/library/gg622935(v=vs.85).aspx
and http://cvlist.ie.microsoft.com/edge/desktop/1432152749/edgecompatviewlist.xml

Google/Chrome
In the past there was
https://www.chromium.org/Home/chromecompatfaq
Now probably this (but slightly different)
https://www.chromium.org/blink/platform-predictability

**Comment by karlcow** [comment]
Another thought around this. The day webcompat.com disappears for any reasons. the references from the devtools die and needs to be backported to mozilla.org. GoFastly is a Mozilla product and because of it should live on mozilla.org domain at least for its persistence policy.

**Comment by karlcow** [comment]
Related to https://github.com/webcompat/webcompat.com/issues/1539#issuecomment-299064144

**Comment by miketaylr** [comment]
Why not on MDN?

**Comment by miketaylr** [comment]
> Why not on MDN?

My original thoughts were the wiki, but that might feel less official and pretty than MDN.

**Comment by zoepage** [comment]
As far as I understood @denschub in our call, MDN didn't seem like a well fitting place.
So, maybe having an independent page would make more sense?

The idea behind adding the landing page to webcompat.com was to decrease the development time as it might use the same functionality (as loading GH issues, labels etc etc) as we do with webcompat.com.

**Comment by karlcow** [comment]
Currently the message in the console in the Developer Tools already lands on MDN. For example the security messages about mixed content blocking etc.
https://developer.mozilla.org/en-US/docs/Tools/Web_Console/Console_messages#Security

The why is because GoFaster is mozilla only. Also another reason in terms of persistence policy. If one day we (webcompat.com and/or WebCompat Mozilla team) do not exist anymore at Mozilla, the gofaster tool might continue to exist on its own and this needs to be handle by Mozilla employees.  

But maybe I misunderstood the purpose of the landing pages.

**Comment by zoepage** [comment]
@karlcow no, you didn't. :) It's a valid argument. 

@denschub Let's see, if we can find a different solution? Happy to help!

**Comment by denschub** [comment]
> Why not on MDN?

I am not sure if we can use GET parameters on MDN to distinguish between the different override types.

@a2sheppy, I had a quick look around but was not able to find something. Do you know how/if it is possible to use a url parameter/slug for toggling between different text paragraphs? Imagine this as if we are writing platform specific docs, with a fixed and known set of different values, and some parts of the text replaced by something else in some cases.

**Comment by karlcow** [comment]
> I am not sure if we can use GET parameters on MDN to distinguish between the different override types.

@denschub could you explain. I'm missing a piece of the puzzle. Taking your suggestion in your initial comment. I was thinking the full thing would be like this:

> The user agent for this request has been overridden to ensure Web Compatibility. [More information](https://developer.mozilla.org/en-US/docs/Web_Compatibility/Override/0001)

With a unique page for each individual override. Explaining when it was put in place, when it was removed (if removed) and with links to a meta page explaining what is about the WebCompat GoFaster, how is it working how to deactivate it/activate it.

**Comment by denschub** [comment]
Ah! Initially, I was thinking about a more generic text like "We changed some CSS attributes to make the site look correctly" or "We changed the user agent for this site to make it behave correctly", not a detailed per-case explanation. So I was thinking about writing a text for each case and make them toggleable via a parameter or something.

But you are right, we could create a page per case. That would solve my question to Sheppy, but not sure if it's a good idea to create (possibly) a lot of pages. :)
